src/index.ml

```diff
@@ -876,8 +903,16 @@ struct
 
   (** {1 Replace} *)
 
+  let yield t =
+    let log = Option.get t.log in
+    (* yield every `t.replace_yield_frequency`, when it exists. *)
+    Option.iter
+      (fun i -> if Tbl.length log.mem mod i = 0 then Thread.yield ())
+      t.replace_yield_frequency
+
   let replace' ?hook ?(overcommit = false) t key value =
     let t = check_open t in
+    yield t;
```

Comment:
Since this function looks like `Lwt_unix.auto_yield`, we could name it the name way.

```suggestion
  let auto_yield t =
    let log = Option.get t.log in
    (* yield every `t.replace_yield_frequency`, when it exists. *)
    Option.iter
      (fun i -> if Tbl.length log.mem mod i = 0 then Thread.yield ())
      t.replace_yield_frequency

  let replace' ?hook ?(overcommit = false) t key value =
    let t = check_open t in
    auto_yield t;
```